intelmq/bots/parsers/shadowserver/parser.py

```diff
@@ -134,6 +136,8 @@ def parse_line(self, row, report):
         # extra if an add operation failed
         for item in conf.get('optional_fields'):
             intelmqkey, shadowkey = item[:2]
+            if shadowkey not in fields:  # key does not exist in data (not even in the header)
```

Comment:
I wonder if it wouldn't be appropriate to write at least a warning to
the log file, as this situation (an key expected by the parser
configuratoin is not presend in the feed data) is something an admin
(or developer) might want to know about...